python/sparkdl/param/converters.py

```diff
@@ -0,0 +1,155 @@
+# Copyright 2017 Databricks, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import six
+
+import tensorflow as tf
+
+from pyspark.ml.param import TypeConverters
+
+import sparkdl.graph.utils as tfx
+import sparkdl.utils.keras_model as kmutil
+
+__all__ = ['SparkDLTypeConverters']
+
+
+class SparkDLTypeConverters(object):
+    """
+    .. note:: DeveloperApi
+
+    Factory methods for common type conversion functions for :py:func:`Param.typeConverter`.
+    These methods are similar to :py:class:`spark.ml.param.TypeConverters`.
+    They provide support for the `Params` types introduced in Spark Deep Learning Pipelines.
+    """
+    @staticmethod
+    def toTFGraph(value):
+        if isinstance(value, tf.Graph):
+            return value
+        else:
+            raise TypeError("Could not convert %s to TensorFlow Graph" % type(value))
+
+    @staticmethod
+    def asColumnToTensorNameMap(value):
+        """
+        Convert a value to a column name to :py:obj:`tf.Tensor` name mapping
+        as a sorted list of string pairs, if possible.
+        """
+        if isinstance(value, dict):
+            strs_pair_seq = []
+            for _maybe_col_name, _maybe_tnsr_name in value.items():
+                # Check if the non-tensor value is of string type
+                _col_name = _get_strict_col_name(_maybe_col_name)
+                # Check if the tensor name is actually valid
+                _tnsr_name = _get_strict_tensor_name(_maybe_tnsr_name)
+                strs_pair_seq.append((_col_name, _tnsr_name))
+
+            return sorted(strs_pair_seq)
+
+        err_msg = "Could not convert [type {}] {} to column name to tf.Tensor name mapping"
+        raise TypeError(err_msg.format(type(value), value))
+
+    @staticmethod
+    def asTensorNameToColumnMap(value):
+        """
+        Convert a value to a :py:obj:`tf.Tensor` name to column name mapping
+        as a sorted list of string pairs, if possible.
+        """
+        if isinstance(value, dict):
+            strs_pair_seq = []
+            for _maybe_tnsr_name, _maybe_col_name in value.items():
+                # Check if the non-tensor value is of string type
+                _col_name = _get_strict_col_name(_maybe_col_name)
```

Comment:
These _get_strict... functions don't actually convert anything, so we don't need to call functions that looks like they do. Instead of the _get_strict_... functions, I'd create two functions: `_check_is_str` and `_check_is_tensor_name` that throw errors if the types are not right (the latter can use the former inside). Then this section becomes (from line 71-75):
```
for tensor_name, col_name in value.items():
  _check_is_str(col_name)
  _check_is_tensor_name(tensor_name)
```